web-app/src/app/store/profile-reducer.ts

```diff
@@ -9,14 +9,15 @@ import {
 import { type NavigateFunction } from 'react-router-dom';
 import { type UserCredential } from 'firebase/auth';
 
-interface UserProfileState {
+export interface UserProfileState {
```

Comment:
question: Why do we need to export this internal state? If required, I suggest moving the interface to a common file like, `types.ts`.

Reply:
removed, no need to export internal state